### ISSUE TITLE
Discussion: Offcanvas event API change for svelte 5

### ISSUE THREAD

**Issue by RoryDuncan** [sveltestrap/sveltestrap#99]
Hi Sveltestrap team!

#### TL;DR

Consider changing the implementation of `Offcanvas` in Svelte 5 to **avoid** firing events on mount.

## Discussion

I've noticed a detail with the `Offcanvas` component that I'd like to discuss: it currently fires events upon mounting.
- When `isOpen` is `true` on mount, it triggers `opening` and `open` events.
- When `isOpen` is `false`, it triggers `closing` and `close` events.

This behavior stems from [this segment of code](https://github.com/sveltestrap/sveltestrap/blob/4fec0e1b18419ddc0b82aee6210838b37affad01/src/Offcanvas/Offcanvas.svelte#L140C3-L150C4), which reacts when the `element` is created:

```js
  $: if (element) {
    isOpen = isOpen;
    isTransitioning = true;

    dispatch(isOpen ? 'opening' : 'closing');

    setTimeout(() => {
      isTransitioning = false;
      dispatch(isOpen ? 'open' : 'close');
    }, getTransitionDuration(element));
  }
```

And, [you can observe it within this repl](https://svelte.dev/playground/5fd0b96f564c4cd3b274977904a2a5d8?version=5.1.3).

### Opinion

I think the `Offcanvas` component should not trigger `open`, `opening`, `close`, or `closing` events upon mounting.

Addressing this would be a breaking change, so it might be worth considering this adjustment alongside the migration to Svelte 5. 

Also, I think a re-implementation likely would resolve #68.
